cmd/translate/main.go

```diff
@@ -1,3 +1,58 @@
 package main
 
-func main() {}
+import (
+	"fmt"
+	"os"
+
+	"github.com/spf13/cobra"
+	"github.com/spf13/viper"
+)
+
+var cfgFile string
+
+// rootCmd represents the base command when called without any subcommands.
+var rootCmd = &cobra.Command{
+	Use:   "translate",
+	Short: "Enables translation for Cloud-native systems",
+	Long:  `Enables translation for Cloud-native systems`,
+	// Uncomment the following line if your bare application
+	// has an action associated with it:
+	// Run: func(cmd *cobra.Command, args []string) { },
+}
+
+func main() {
+	Execute()
+}
+
+// Execute adds all child commands to the root command and sets flags appropriately.
+// This is called by main.main(). It only needs to happen once to the rootCmd.
+func Execute() {
+	if err := rootCmd.Execute(); err != nil {
+		os.Exit(1)
+	}
+}
+
+func init() {
+	cobra.OnInitialize(initConfig)
+	rootCmd.PersistentFlags().StringVar(&cfgFile, "config", "", "config file (default is translate/translate.yml)")
+}
+
+// initConfig reads in config file and ENV variables if set.
+func initConfig() {
+	if cfgFile != "" {
+		// Use config file from the flag.
+		viper.SetConfigFile(cfgFile)
+	} else {
+		// Sets default config.
+		viper.SetConfigFile("translate.yml")
```

Comment:
wont work, as `yml` file is antoher dir, I should be at same dir as `main.go`, or give full path, but first option should be more preferrable

Reply:
okay, I am not sure about this anymore. It takes the directory you are executing it.

So if you are currently in root dir, and `translate.yml` is in `cmd/translate` and `run go run cmd/translate/main.go` then it **will not** find `translate.yml`

but if you are in `cmd/translate` dir, and `translate.yml` is in `cmd/translate` then running `go run main.go`, **will** find `translate.yml`

For me storing `translate.yml` in `cmd/translate` is a logical decision, so I would suggest remaining it there, and just set the absolute path as it was before, e.g. `dir, err := os.Getwd()` ...

Reply:
Yes, I was able to reproduce mentioned issue I will commit fix now with your proposal (to use os.Getwd). And I also asked question to Janis about which option is correct or mb we should support both of them. (go run cmd/translate/main.go or go run main.go)

Reply:
to run default config @VladislavsPerkanuks go run ../cmd/translate/main.go.